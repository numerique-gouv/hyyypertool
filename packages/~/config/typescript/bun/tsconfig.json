{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Bun",
  "extends": "../base/tsconfig.json",
  "compilerOptions": {
    "types": ["bun-types"]
  }
}
