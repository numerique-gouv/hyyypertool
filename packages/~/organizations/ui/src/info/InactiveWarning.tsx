//

import { type Organization } from "@~/organizations.lib/entities/Organization";
import { type JSX } from "hono/jsx";

//

type Props = JSX.IntrinsicElements["section"] & {
  organization: Pick<Organization, "cached_est_active">;
};

export function InactiveWarning(props: Props) {
  const { organization, ...section_props } = props;

  if (organization.cached_est_active) {
    return null;
  }

  return (
    <section {...section_props}>
      <div class="fr-alert fr-alert--warning">
        <i class="bi bi-exclamation-triangle-fill" />
        <h3 class="fr-alert__title">
          Attention : cette organisation a cessé son activité.
        </h3>
      </div>
    </section>
  );
}
