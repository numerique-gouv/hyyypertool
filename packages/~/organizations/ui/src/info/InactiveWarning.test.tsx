//

import { expect, test } from "bun:test";
import { Hono } from "hono";
import { jsxRenderer } from "hono/jsx-renderer";
import { format } from "prettier";
import { InactiveWarning } from "./InactiveWarning";

test("render inactive warning", async () => {
  const response = await new Hono()
    .get("/", jsxRenderer(), ({ render }) =>
      render(
        <InactiveWarning
          organization={{
            cached_est_active: true,
          }}
        />,
      ),
    )
    .request("/");
  expect(response.status).toBe(200);
  expect(
    await format(await response.text(), { parser: "html" }),
  ).toMatchSnapshot();
});

test("render nothing", async () => {
  const response = await new Hono()
    .get("/", jsxRenderer(), ({ render }) =>
      render(
        <InactiveWarning
          organization={{
            cached_est_active: false,
          }}
        />,
      ),
    )
    .request("/");
  expect(response.status).toBe(200);
  expect(
    await format(await response.text(), { parser: "html" }),
  ).toMatchSnapshot();
});
