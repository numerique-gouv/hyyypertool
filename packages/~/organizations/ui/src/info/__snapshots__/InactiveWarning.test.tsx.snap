// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`render inactive warning 1`] = `
"<!doctype html>
"
`;

exports[`render nothing 1`] = `
"<!doctype html>
<section>
  <div class="fr-alert fr-alert--warning">
    <i class="bi bi-exclamation-triangle-fill"></i>
    <h3 class="fr-alert__title">
      Attention : cette organisation a cessé son activité.
    </h3>
  </div>
</section>
"
`;
